pycoal/mineral.py

```diff
@@ -10,225 +10,79 @@
+        for x in xrange(M):
 
-            # TODO convert units ?
-            # see https://groups.google.com/d/msg/coal-capstone/6oordALy0dA/a_6VIuWbBAAJ
+            for y in xrange(N):
 
-            # define a linear transformation to scale the pixel if necessary
-            # the scalars would presumably be read from metadata
-            #transform = spectral.transforms.LinearTransform(???)
+                # read the pixel from the file
+                pixel = data[x,y]
 
-            # transform it
-            #transformedPixel = transform(pixel)
-
-            # classify and store it in the classified image
-            classifiedImage[imageIndex] = self.classifyPixel(pixel, classifier, wavelengthList)
-
-            imageIndex += 1
-        # save the classified image to a file
-        spectral.io.envi.save_classification(classifiedImageFilename, classifiedImage)
+                # if it is not a no data pixel
+                if not numpy.isclose(pixel[0], -0.005) and not pixel[0]==-50:
 
-    def classifyImages(self, imageFilenames, classifier, classifiedImageFilenames):
-        """
-        Classify a set of AVIRIS images using a trained classifier and save the results to files.
-        Args:
-           imageFilenames (str): the images to be classified
-           classifier (PerceptronClassifier): the classifier
-           classifiedImageFilenames (str): the files which to save the classified images
-        Returns:
-            None
-        """
+                    # resample the pixel ignoring NaNs from target bands that don't overlap
+                    # TODO fix spectral library so that bands are in order
+                    resampledPixel = numpy.nan_to_num(resample(pixel))
 
-        # for each image in a set of images
-        for imageFilename,classifiedImageFilename in zip(imageFilenames,classifiedImageFilenames):
+                    # calculate spectral angles
+                    angles = spectral.spectral_angles(resampledPixel[numpy.newaxis,
+                                                                     numpy.newaxis,
+                                                                     ...],
+                                                      self.library.spectra)
 
-            # classify the image and save it to a file
-            self.classifyImage(imageFilename, classifier, classifiedImageFilename)
+                    # get classification
+                    classified[x,y] = numpy.argmin(angles) + 1
 
+        # save the classified image to a file
+        spectral.io.envi.save_classification(classifiedFilename,
+                                             classified,
+                                             class_names=['No data']+self.library.names,
+                                             metadata={
+                                                 'data ignore value': 0,
+                                                 'description': 'Mineral classified image.',
```

Comment:
i added the version number to `__init.py__`:
```
[ec2-user@ip-172-31-28-45 pycoal]$ git show HEAD
commit b6a2d43a179a42de2baf0ed1409a43391771575d
Author: Taylor Alexander Brown <browtayl@oregonstate.edu>
Date:   Fri Mar 17 03:23:20 2017 +0000

    add version number to image header

diff --git a/pycoal/__init__.py b/pycoal/__init__.py
index e69de29..7c96902 100644
--- a/pycoal/__init__.py
+++ b/pycoal/__init__.py
@@ -0,0 +1 @@
+version = "0.2"
diff --git a/pycoal/mineral.py b/pycoal/mineral.py
index 8911296..4acd2b2 100644
--- a/pycoal/mineral.py
+++ b/pycoal/mineral.py
@@ -12,6 +12,7 @@
 
 import numpy
 import spectral
+import pycoal
 
 class MineralClassification:
 
@@ -84,6 +85,6 @@ class MineralClassification:
                                              class_names=['No data']+self.library.names,
                                              metadata={
                                                  'data ignore value': 0,
-                                                 'description': 'Mineral classified image.',
+                                                 'description': 'PyCOAL '+pycoal.version+' mineral classified image.',
                                                  'map info': image.metadata.get('map info')
                                              })
```
now the classified header file looks like this:
```
[ec2-user@ip-172-31-28-45 browtayl]$ head f080702t01p00r08rdn_c_sc01_ort_img_class_test.hdr 
ENVI
description = {
  PyCOAL 0.2 mineral classified image.}
samples = 10
lines = 10
bands = 1
header offset = 0
file type = ENVI Classification
data type = 12
interleave = bip
```